docs/languages/csharp/expression-trees-execution.md

```diff
@@ -0,0 +1,194 @@
+or a type derived from `LambdaExpression` can be converted to IL.
+The expression type `Expression<TDelegate> where TDelegate : delegate`
+is the only concreted example in the .net Core libraries. It's used
+to represent an expression that maps to any delegate type. Because
+this type maps to a delegate type, the .NET framework can examine
+the expression, and generate IL for an appropriate delegate that
+matches the signature of the lambda expression. 
+
+In most cases, theis creates a simple mapping between an expression,
+and its corresponding delegate. For example, an expression tree that
+is represented by `Expression<Func<int>>` would be converted to a delegate
+of the type `Func<int>`. For a lambda expression with any return type
+and argument list, there exists a delegate type that is the target type
+for the executable code represented by that lamdba expression.
+
+The `LamdbaExpression` type contains `Compile` and `CompileToMethod`
+members that you would use to convert an expression tree to executable
+code. The `Compile` method creates a delegate. The `ConmpileToMethod`
+method updates a `MethodBuilder` object with the IL that represents
+the compiled output of the expression tree. Note that `CompileToMethod`
+is only available on the full desktop framework, not on the 
+.NET Core framework.
+
+Optionally, you can also provide a `DebugInfoGenerator` that will
+receive the symbol debugging information for the generated delegate
+object. This enables you to convert the expression tree into a
+delegate object, and have full debugging information about the
+generated delegate.
+
+You would convert an expression into a delegate using the following
+code:
+
+```cs
+Expression<Func<int>> add = () => 1 + 2;
+var func = add.Compile(); // Create Delegate
+var answer = func(); // Invoke Delegate
+Console.WriteLine(answer);
+```
+
+Notice that the delegate type is based on the expression type. You must
+know the return type and the argument list if you want to use the
+delegate object in a strongly typed manner. The `LambdaExpression.Compile()`
+method returns the `Delegate` type. You will have to cast it to the correct
+delegate type to have any compile-time tools check the argument list of
+return type.
+
+## Execution and Lifetimes
+
+You execute the code by invoking the delegate created
+you called `LamdbaExpression.Compile()`.
+
+That delegate represents the code in the expression tree. You can
+retain the handle to that delgate and invoke it later. You don't need
+to compile the expression tree each time you want to execute the code
+it represents. (Remember that expression trees are immutable, and
+compiling the same expression tree later will create a delegate that
+executes the same code.)
+
+I will caution you against trying to create any more sophisticated
+caching mechanisms to increase performance by avoiding unnecessary
+compile calls. It's quite a difficult and painstaking operation to
```

Comment:
I'd frame this a bit more positively if possible.  We don't want to send the message that this is an API that's hard to use (which may be the case, but is something that should be reserved for API reviews).